source/common/network/address_impl.cc

```diff
@@ -130,7 +139,15 @@ int Ipv4Instance::connect(int fd) const {
 }
 
 int Ipv4Instance::socket(SocketType type) const {
-  return ::socket(AF_INET, flagsFromSocketType(type), 0);
+  int fd = ::socket(AF_INET, flagsFromSocketType(type), 0);
+  RELEASE_ASSERT(fd != -1);
+
+#ifdef __APPLE__
+  // Cannot set SOCK_NONBLOCK as a ::socket flag
```

Comment:
There are perf implications for this (multiple system calls vs. 1), so in this case I would be in favor of the ifdef. If there is a way we can make it less visually annoying that's fine.